pilot/pkg/model/destination_rule.go

```diff
@@ -53,21 +53,33 @@ func (ps *PushContext) mergeDestinationRule(p *consolidatedDestRules, destRuleCo
 		// `addRuleToProcessedDestRules` determines if the incoming destination rule would become a new unique entry in the processedDestRules list.
 		addRuleToProcessedDestRules := true
 		for _, mdr := range mdrList {
+			if exportToSet.Equals(mdr.exportTo) {
+				addRuleToProcessedDestRules = false
+			} else {
+				// If the exportTo totally not equal, skip merging
+				// If one is subset of the other, continue merging
+				if len(exportToSet.Difference(mdr.exportTo)) > 0 && len(mdr.exportTo.Difference(exportToSet)) > 0 {
+					continue
+				}
+			}
+
 			existingRule := mdr.rule.Spec.(*networking.DestinationRule)
 			bothWithoutSelector := rule.GetWorkloadSelector() == nil && existingRule.GetWorkloadSelector() == nil
 			bothWithSelector := existingRule.GetWorkloadSelector() != nil && rule.GetWorkloadSelector() != nil
 			selectorsMatch := labels.Instance(existingRule.GetWorkloadSelector().GetMatchLabels()).Equals(rule.GetWorkloadSelector().GetMatchLabels())
-
 			if bothWithSelector && !selectorsMatch {
 				// If the new destination rule and the existing one has workload selectors associated with them, skip merging
 				// if the selectors do not match
+				addRuleToProcessedDestRules = true
 				continue
 			}
 			// If both the destination rules are without a workload selector or with matching workload selectors, simply merge them.
 			// If the incoming rule has a workload selector, it has to be merged with the existing rules with workload selector, and
 			// at the same time added as a unique entry in the processedDestRules.
 			if bothWithoutSelector || (rule.GetWorkloadSelector() != nil && selectorsMatch) {
-				addRuleToProcessedDestRules = false
+				addRuleToProcessedDestRules = addRuleToProcessedDestRules || false
```

Comment:
What's the intention with this logic? It seems to assume that `addRuleToProcessedDestRules` won't be set to `false` if not `true`. Or is this done for readability reasons; to make it clear that it's being set to false here?

If so. Logically, is this the same as the following?
```go
if bothWithoutSelector || (rule.GetWorkloadSelector() != nil && selectorsMatch) && !addRuleToProcessedDestRules {
    addRuleToProcessedDestRules = false
} else {
    addRuleToProcessedDestRules = true
}
```

Reply:
Yes, equivalent